main.go

```diff
@@ -24,6 +28,122 @@ type AvailableReleaseResponse struct {
 
 func main() {
 
+	if len(os.Args) == 1 {
+		InitHipo()
+	} else if len(os.Args) == 2 {
+		destPath := downloadFile(os.Args[1])
+		executeFile(destPath)
+	}
+}
+
+func downloadFile(Args string) string {
+
+	parts := strings.Split(Args, ":")
+	if len(parts) != 3 {
+		fmt.Println("Invalid coordinate format. Use <group:artifact:version>")
+		return ""
+	}
+
+	// Go to link
+	groupID := parts[0]
+	artifactID := parts[1]
+	version := parts[2]
+
+	groupPath := strings.ReplaceAll(groupID, ".", "/")
+	artifactFilename := fmt.Sprintf("%s-%s.jar", artifactID, version)
+	url := fmt.Sprintf("https://repo1.maven.org/maven2/%s/%s/%s/%s", groupPath, artifactID, version, artifactFilename)
+
+	homeDir, err := os.UserHomeDir()
+
+	if err != nil {
+		fmt.Println("Error:", err)
+		return ""
+	}
+
+	// Create directory to save the JAR file
+	destDir := filepath.Join(homeDir, ".hipo", groupPath, artifactID, version)
+	err2 := os.MkdirAll(destDir, 0755)
+	if err2 != nil {
+		fmt.Printf("Error creating directory: %v\n", err)
+		return ""
+	}
+
+	destPath := filepath.Join(destDir, artifactFilename)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Printf("failed to make GET request: %v", err)
+		return ""
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		fmt.Printf("failed to download file: status code %d", resp.StatusCode)
+		return ""
+	}
+
+	out, err := os.Create(destPath)
+	if err != nil {
+		fmt.Printf("failed to create file: %v", err)
+		return ""
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		fmt.Printf("failed to copy response body to file: %v", err)
+		return ""
+	}
+
+	// fmt.Printf("Downloaded to %s\n", destPath)
+	return destPath
+}
+
+func executeFile(jarFilePath string) {
+
+	homeDir, err := os.UserHomeDir()
+
+	if err != nil {
+		fmt.Println("Error:", err)
+		return
+	}
+
+	jreParentDir := filepath.Join(homeDir, ".hipo", "jre")
+
+	// List the directories in the parent JRE directory
+	dirs, err := os.ReadDir(jreParentDir)
+	if err != nil {
+		fmt.Println("Error reading JRE directory:", err)
+		return
+	}
+
+	if len(dirs) == 0 {
+		fmt.Println("No JDK directories found")
+		return
+	}
+
+	// Assume there is only one directory and get its name
+	jdkDir := dirs[0].Name()
+
+	// Path to the java executable
+	javaExecPath := filepath.Join(homeDir, ".hipo", "jre", jdkDir, "bin", "java")
+
+	cmd := exec.Command(javaExecPath, "-jar", jarFilePath)
+
+	// Set the command's standard output and standard error
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		fmt.Println("Error running the Java command:", err)
+		return
+	}
+
+	//fmt.Println("Java application ran successfully")
+}
+
+func InitHipo() {
 	hipoHome, done := PrepareHipoHome()
 
 	if !done {
```

Comment:
if you have an empty `.hipo` directory the rest of the code doesn't work. Method should be like:

```go
hipoHome, exists := PrepareHipoHome()
```